src/mpid/ch4/src/ch4r_recvq.c

```diff
@@ -41,14 +41,14 @@ int MPIDIG_recvq_init(void)
     MPIR_T_register_source("RECVQ", "active message receive queue", MPI_T_SOURCE_ORDERED, NULL, -1,
                            -1, &source_index);
 
-    MPI_Datatype array_of_datatypes[1] = { MPI_INT };
-    MPI_Aint array_of_displacements[1] = { 0 };
+    MPI_Datatype array_of_datatypes[2] = { MPI_INT, MPI_INT };
+    MPI_Aint array_of_displacements[2] = { 0, 4 };
```

Comment:
This commit should include the exposure of this information at the actual event sites. Otherwise codes that attempt to read the information could crash (or get garbage values).

In general, commits should work as independent units. I.e. each one should compile and function on its own. This one is incomplete in that regard.

Reply:
@raffenet this should be fixed now. I now have two commits (one for the test case and one for the modifications to the RECVQ events source code).

Reply:
Looks good 👍